go-controller/pkg/node/gateway_udn.go

```diff
@@ -359,11 +359,23 @@ func (udng *UserDefinedNetworkGateway) AddNetwork() error {
 	if err = udng.vrfManager.AddVRFRoutes(vrfDeviceName, routes); err != nil {
 		return fmt.Errorf("could not add VRF %s routes for network %s, err: %v", vrfDeviceName, udng.GetNetworkName(), err)
 	}
+
+	isNetworkAdvertised := util.IsPodNetworkAdvertisedAtNode(udng.NetInfo, udng.node.Name)
```

Comment:
so now I am getting a bit confused, are we going to introduce the notion of BGP into `AddNetwork` itself?
to answer myself - yes because if the RA is created before the CUDN @kyrtapz want's that info of RA being present being done during CUDN creation I guess?

I guess this is because `doReconcile` is not getting called when a normal CUDN network get's added freshly since there is no previous state.

its a bit sad that now we have two different event ordered paths for advertised udns add -> this function AddNetwork and non-advertised udn adds followed by update to make them advertised -> doReconcile

Maybe AddorUpdateNetwork was the better design choice but that is not related to this specific PR. I guess we need to make gateway code also all level driven at some point.

Reply:
I assume no problem is there on delete ordering ?

- RA is deleted which calls doReconcile which puts flows back to UDN-y and then CUDN is deleted so delNetwork is called
- CUDN is deleted alltogether so delNetwork is called .. hm in this case you correctly already remove the subnets i guess

Reply:
Delete is covered already yeah.
We do need to handle advertisement config on Add as well to cover the case you mentioned. this was already done for IP rules but was less explicit(I refactored that in the second commit).